core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -155,7 +155,8 @@ public final ClientStream newStream(MethodDescriptor<?, ?> method, Metadata head
       }
       if (picker != null) {
         while (true) {
-          PickResult pickResult = picker.pickSubchannel(callOptions.getAffinity(), headers);
+          PickResult pickResult = picker.pickSubchannel(
+              new PickSubchannelArgsImpl(method, headers, callOptions));
```

Comment:
Tried to do a quick refactor, some tests were failing. Is it ok to address this in a follow up PR?